### PR TITLE
GUVNOR-3350: [Stunner] Toolbox views do not enable/disable

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-3350

@romartin I noticed this when doing some changes for DMN (hence there's some leakage of some access modifiers too.. if you want them reverted and included in my DMN PR then just say so).

### CODE REVIEW

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/toolbar/impl/AbstractToolbar.java [hunk 1]
```diff
@@ -101,10 +113,13 @@ public void destroy() {
...
-        return items.stream()
-                .filter(command::equals)
+        return items
+                .entrySet()
+                .stream()
+                .filter(e -> e.getKey().equals(command))
                 .findFirst()
-                .orElse(null);
+                .orElse(null)
+                .getValue();
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/toolbar/impl/AbstractToolbar.java]: hey @manstis , quick looking at this seems that a potential NPE can be produced here? I mean for `orElse(null).getValue()`?

**Comment by manstis** [comment]
@romartin Good spot! Fixed and test added.. these reviews are worth their weight in gold!

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/toolbar/impl/AbstractToolbar.java [hunk 2]
```diff
@@ -101,10 +114,15 @@ public void destroy() {
...
-        return items.stream()
-                .filter(command::equals)
-                .findFirst()
-                .orElse(null);
+        final Optional<Map.Entry<ToolbarCommand<? super S>, AbstractToolbarItem<S>>> item = items
+                .entrySet()
+                .stream()
+                .filter(e -> e.getKey().equals(command))
+                .findFirst();
+        if (item.isPresent()) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/toolbar/impl/AbstractToolbar.java]: yeah thanks @manstis ! jBTW ust for the records, you can also you the `map` operation here to solve this, as the map only gets executed if the optional is present :)

**Comment by manstis** [comment]
@romartin Changed to use ```map(..)``` as suggested thanks (I just knew they'd be a way to do it in one go!)

**Comment by manstis** [comment]
Jenkins please retest this my dear friend.

**Comment by manstis** [comment]
@hasys Just waiting on your approval please :-)

**Comment by manstis** [comment]
Thanks for the advice @hasys to be honest I commit from the CLI... but perhaps it's time I changed!

**Comment by Christopher-Chianelli** [comment]
You can also auto-format code @hasys! If I recall, I did a pull request which talked about this where all the instructions for new contributors are. Been commiting from InteliJ ever since I found about this feature. Nonetheless, thanks for the advice @hasys!

**Pull request merged**